src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -77,50 +78,104 @@ class WorkspaceNewRoomPage extends React.Component {
     }
 
     /**
-     * Called when a workspace is selected. Also calls checkAndModifyRoomName,
-     * which displays an error if the given roomName exists on the newly selected workspace.
-     * @param {String} policyID
+     * Called when the "Create Room" button is pressed.
      */
-    onWorkspaceSelect(policyID) {
-        this.setState({policyID});
-        this.checkAndModifyRoomName(this.state.roomName);
+    onSubmit() {
+        if (!this.validate()) {
+            return;
+        }
+        Report.createPolicyRoom(
+            this.state.policyID,
+            this.state.roomName,
+            this.state.visibility,
+        );
     }
 
     /**
-     * Called when the "Create Room" button is pressed.
+     * Called when the onSubmit() method is triggered.
+     * which validates the form fields.
+     * @returns {Boolean}
      */
-    onSubmit() {
-        Report.createPolicyRoom(this.state.policyID, this.state.roomName, this.state.visibility);
+    validate() {
+        const roomName = this.state.roomName.trim();
+        const policyID = this.state.policyID;
+        const visibility = this.state.visibility;
+        const errors = {};
+
+        const isExistingRoomName = _.some(
+            _.values(this.props.reports),
+            report => report && report.policyID === this.state.policyID
+            && report.reportName === this.modifyRoomName(roomName),
```

Comment:
If we already modified the room name before storing it in the state then why do we need to modify it here again?

Reply:
Ohh yaa actually. I got confused haha.